pkg/core/logger/logger.go

```diff
@@ -72,6 +73,31 @@ func NewWithoutConfig(name string) *Logger {
 	return logp.NewLogger(name)
 }
 
+// NewInMemory returns a new in-memory logger along with the buffer to which i
+// logs.
+// This logger utilizes a console format; for details, refer to
+// ecszap.ECSCompatibleEncoderConfig and logp.ConsoleEncoderConfig.
+func NewInMemory(selector string) (*Logger, *bytes.Buffer) {
+	buff := bytes.Buffer{}
+
+	encoderConfig := ecszap.ECSCompatibleEncoderConfig(logp.ConsoleEncoderConfig())
```

Comment:
Can we inject the `logp.ConsoleEncoderConfig()` configuration as parameter? Just in case this needs to be changed depending on the call location/ additional usages
Even better we could receive our log configuration, minus the output part